### PR TITLE
Add sphinx-codeautolink package

### PR DESCRIPTION
**Pull request by Zeitsperre**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by Zeitsperre** [comment]
@felix-hilden Can you please confirm in a comment that you wish to be listed as a maintainer? Thanks!

File: recipes/sphinx-codeautolink/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+requirements:
+  host:
+    - python >=3.6
+    - pip
+  run:
+    - python >=3.6
+    - sphinx >=3.2.0
+    - beautifulsoup4
+    - dataclasses
```
Comment (felix-hilden) [inline recipes/sphinx-codeautolink/meta.yaml:25]: I don't suppose there's a way to make Python version dependent requirements? If not, go for 3.7+ and drop dataclasses IMO 😄

**Review by felix-hilden — APPROVED** [review]
Thanks for making it happen! I'm 100 % okay with my package being in conda-forge. If I must, I will agree to being a "maintainer" as well, but I'd prefer not to, since I don't really know what that entails 😅 I don't know which one you wanted me to confirm.

File: recipes/sphinx-codeautolink/meta.yaml (hunk 1 above)
Comment (Zeitsperre) [reply recipes/sphinx-codeautolink/meta.yaml:25]: From what I recall reading, `dataclasses` will only be explicitly added to Python3.6 builds. It's in the standard library for 3.7+, so I imagine that once conda-forge drops Python3.6 support completely, this won't be needed anymore.

**Comment by Zeitsperre** [comment]
@conda-forge/help-python, this is good for review, thanks!

**Comment by dopplershift** [comment]
@felix-hilden If you prefer not to be a maintainer (for conda-forge), @Zeitsperre can remove your name from the maintainers listed here. Being the maintainer means you have the permissions to merge updates to the recipe (e.g. for build infrastructure changes, version updates--which our bot picks up automatically)--and it means you'll receive notifications for PRs for these.

**Comment by felix-hilden** [comment]
If you're ok with being the sole maintainer yourself @Zeitsperre, this seems alright! I wouldn't be (pro)active.

Thanks for taking this in though <3

**Comment by Zeitsperre** [comment]
@felix-hilden Not an issue at all. I don't mind!

**Review by dopplershift — APPROVED** [review]

**Pull request merged**